src/charm.py

```diff
@@ -194,6 +201,24 @@ def _validate_pebble_plan(self, container):
         except pebble.ConnectionError:
             return False
 
+    def _validate_self_registration_role(self):
+        """Determine allowed Superset roles.
+
+        Raises:
+            ValueError: in case role value is not allowed.
+        """
+        uri = self._state.sql_alchemy_uri
+        sql = SQL_AB_ROLE
+
+        allowed_roles = query_metadata_database(uri, sql)
+        if not allowed_roles:
+            allowed_roles = DEFAULT_ROLES
+        role = self.config["self-registration-role"]
+        if role not in allowed_roles:
```

Comment:
wanna make this case insensitive? like either upper everything or lower everything.

Reply:
The value itself is case sensitive. You can create `Marketing` and `marketing` roles in Superset, which I wouldn't be able to distinguish between. I'll add a note in the `config.yaml`.